Makefile

```
@@ -124,8 +124,16 @@ E2E_REGISTRY_NAME := docker-registry
 E2E_REGISTRY_NAMESPACE := operator-controller-e2e
 
 export REG_PKG_NAME := registry-operator
-export REGISTRY_ROOT := $(E2E_REGISTRY_NAME).$(E2E_REGISTRY_NAMESPACE).svc:5000
-export CATALOG_IMG := $(REGISTRY_ROOT)/e2e/test-catalog:e2e
+export LOCAL_REGISTRY_HOST := $(E2E_REGISTRY_NAME).$(E2E_REGISTRY_NAMESPACE).svc:5000
+export CLUSTER_REGISTRY_HOST := localhost:30000
+export V1_IMAGE_TAG := v1
+export V2_IMAGE_TAG := v2
+export LATEST_IMAGE_TAG := latest
+export CATALOG_IMG := $(LOCAL_REGISTRY_HOST)/e2e/test-catalog:$(V1_IMAGE_TAG)
+export UPDATED_CATALOG_IMG := $(LOCAL_REGISTRY_HOST)/e2e/test-catalog:$(V2_IMAGE_TAG)
+export LATEST_CATALOG_IMG := $(LOCAL_REGISTRY_HOST)/e2e/test-catalog:$(LATEST_IMAGE_TAG)
+export CLUSTER_V1_CATALOG_IMG := $(CLUSTER_REGISTRY_HOST)/e2e/test-catalog:$(V1_IMAGE_TAG)
+export CLUSTER_V2_CATALOG_IMG := $(CLUSTER_REGISTRY_HOST)/e2e/test-catalog:$(V2_IMAGE_TAG)
```

Two comments here: 
1. I don't think we need envvars for every combo. Let's just concatenate.
2. Let's pick a convention like `<repoName>_<imageName_<tag>` and use that.
3. I don't think we'll need a "latest" variant here anymore. The test that needs it can deal with pushing its own tag local to the test.

So I think all of this can boil down to:
```suggestion
E2E_TESTCATALOG_V1 = e2e/testcatalog:v1
E2E_TESTCATALOG_V2 = e2e/testcatalog:v2
```

And then when we need to reference a fully-qualified image, we would just use, for example:
```
$(LOCAL_REGISTRY_HOST)/$(E2E_TEST_CATALOG_V1)
```

---

but we still need to export the cluster image as `CLUSTER_V1_CATALOG_IMG := $(CLUSTER_REGISTRY_HOST)/$(E2E_TEST_CATALOG_V1)` in the makefile, right?

---

Correct me if I'm wrong @joelanford, but I think you want `less` stuff in the Makefile, and more of the `work` in the tests that need "additional" things. IE instead of exported vars, concatenate them in the go code? So it would be something like:

```
fmt.Sprintf("%s/%s", os.Getenv("LOCAL_REGISTRY_HOST"), os.Getenv("E2E_TEST_CATALOG_V1"))
```

Or did you mean something entirely different.

---

This has been implemented now in the latest commit. Thanks